src/coap_client_zephyr.c

```diff
@@ -0,0 +1,1147 @@
+                recv_expiry = k_uptime_get() + RECV_TIMEOUT;
+                ping_expiry = k_uptime_get() + PING_INTERVAL;
+
+                err = golioth_process_rx(client);
+                if (err) {
+                    LOG_ERR("Failed to receive: %d", err);
+                    client_disconnect(client);
+                }
+            }
+
+            if (fds[POLLFD_MBOX].revents) {
+                if (coap_io_loop_once(client) != GOLIOTH_OK) {
+                    client->end_session = true;
+                }
+            }
+        }
+
+        LOG_INF("Ending session");
+
+        if (client->event_callback && client->session_connected) {
+            client->event_callback(
+                    client, GOLIOTH_CLIENT_EVENT_DISCONNECTED, client->event_callback_arg);
+        }
+        client->session_connected = false;
+
+        // Small delay before starting a new session
+        golioth_sys_msleep(1000);
+    }
+}
+
+static int credentials_set_psk(const struct golioth_psk_credential* psk) {
+    int err;
+
+    err = tls_credential_add(
+        sec_tag_list[0],
+        TLS_CREDENTIAL_PSK_ID,
+        psk->psk_id,
+        psk->psk_id_len);
+    if (err) {
+        LOG_ERR("Failed to register PSK ID: %d", err);
+        return err;
+    }
+
+    err = tls_credential_add(
+        sec_tag_list[0],
+        TLS_CREDENTIAL_PSK,
+        psk->psk,
+        psk->psk_len);
+    if (err) {
+        LOG_ERR("Failed to register PSK: %d", err);
+        return err;
+    }
+
+    return 0;
+}
+
+static int credentials_set_pki(const struct golioth_pki_credential* pki) {
+    int err;
+
+    err = tls_credential_add(
+              sec_tag_list[0],
+              TLS_CREDENTIAL_CA_CERTIFICATE,
+              pki->ca_cert,
+              pki->ca_cert_len);
+    if (err) {
+        return 0;
+    }
+
+    err = tls_credential_add(
+              sec_tag_list[0],
+              TLS_CREDENTIAL_SERVER_CERTIFICATE,
+              pki->public_cert,
+              pki->public_cert_len);
+    if (err) {
+        return 0;
+    }
+
+    err = tls_credential_add(
+              sec_tag_list[0],
+              TLS_CREDENTIAL_PRIVATE_KEY,
+              pki->private_key,
+              pki->private_key_len);
+    if (err) {
+        return 0;
+    }
+
+    return 1;
+}
+
+static int credentials_set(const struct golioth_client_config* config) {
+    const struct golioth_credential *credentials = &config->credentials;
+
+    switch (credentials->auth_type) {
+    case GOLIOTH_TLS_AUTH_TYPE_PSK:
+        return credentials_set_psk(&credentials->psk);
+    case GOLIOTH_TLS_AUTH_TYPE_PKI:
+        return credentials_set_pki(&credentials->pki);
+    }
+
+    return -ENOTSUP;
+}
+
+struct golioth_client* golioth_client_create(const struct golioth_client_config* config) {
+    if (!_initialized) {
+        // Seed the random number generator. Used for token generation.
+        time_t t;
+        golioth_sys_srand(time(&t));
```

Comment:
`golioth_sys_srand()` is a noop when targeting Zephyr (Zephyr automatically seeds the prng), so we can just remove this altogether.

Reply:
Removed that statement!